urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/agent/infocollect/DynamicSwapInfo.java

```diff
@@ -0,0 +1,124 @@
+package com.sinopec.agent.infocollect;
+
+import com.sinopec.agent.share.SharedArea;
+import com.sinopec.utils.LogUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created with IntelliJ IDEA. User: Administrator Date: 13-10-23 Time: 下午2:17
+ * To change this template use File | Settings | File Templates.
+ */
+public class DynamicSwapInfo implements Runnable {
+
+	private static final Log LOG = LogFactory.getLog(DynamicSwapInfo.class);
+
+	private int samplingInterval;
+
+	private volatile boolean running = true;
+
+	public DynamicSwapInfo(int samplingInterval) {
+		this.samplingInterval = samplingInterval;
+	}
+
+	public void getSwapInfoFromProc() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.